CI/windows/build_librist.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+###############################################################################
+# Windows librist cross-compile build script
+###############################################################################
+#
+# This script file can be included in build scripts for Windows or run directly
+#
+###############################################################################
+
+# Halt on errors
+set -eE
+
+_patch_product() {
+    cd "${PRODUCT_FOLDER}"
+
+    step "Apply patches..."
+    if [ ! -f "cross_mingw32.txt" ] || [ ! -f "cross_mingw64.txt" ]; then
+        apply_patch "${CHECKOUT_DIR}/CI/windows/patches/librist/librist.patch" "F7417521457C875B901209F849EFDDB092A2BB6897A2797F22FE843EE5AA2363"
+    fi
+}
+
+_build_product() {
+    cd "${PRODUCT_FOLDER}"
+    BASE_DIR="$(pwd)"
+
+    if [ "${ARCH}" = "x86_64" ]; then
+        step "Configure (x86_64)..."
+        meson setup build_x86_64 \
+            --cross-file cross_mingw64.txt \
+            --buildtype release \
+            --prefix "${BUILD_DIR}" \
+            -Duse_mbedtls=true \
+            -Dbuiltin_cjson=true \
+            -Dtest=false \
+            -Dbuilt_tools=false \
+            -Dhave_mingw_pthreads=true \
+            -Dpkg_config_path="${BUILD_DIR}/lib/pkgconfig"
+
+        step "Build (x86_64)..."
+        ninja -C build_x86_64
+    fi
+
+    if [ "${ARCH}" = "x86" ]; then
+        step "Configure (x86)..."
+        meson setup build_x86 \
+            --cross-file cross_mingw32.txt \
+            --buildtype release \
+            --prefix "${BUILD_DIR}" \
+            -Duse_mbedtls=true \
+            -Dbuiltin_cjson=true \
+            -Dtest=false \
+            -Dbuilt_tools=false \
+            -Dhave_mingw_pthreads=true \
+            -Dpkg_config_path="${BUILD_DIR}/lib/pkgconfig"
+
+        step "Build (x86)..."
+        ninja -C build_x86
+    fi
```

Comment:
You could instead do something like:
```
step "Configure (${ARCH})..."
meson setup build_${ARCH} \
    --cross-file cross_mingw${CMAKE_BITNESS}.txt \
    --buildtype release \
    --prefix "${BUILD_DIR}" \
    -Duse_mbedtls=true \
    -Dbuiltin_cjson=true \
    -Dtest=false \
    -Dbuilt_tools=false \
    -Dhave_mingw_pthreads=true \
    -Dpkg_config_path="${BUILD_DIR}/lib/pkgconfig"

step "Build (${ARCH})..."
ninja -C build_${ARCH}
```

And then define CMAKE_BITNESS in `CI/include/build_support_windows_cross.sh` within `check_archs()`.  Alternatively, leave this as-is, and I can optimize that portion later along with the git stuff.

Reply:
i'll leave as is (sorry but i'm a bit exhausted by the fight with this PR)

Reply:
i've updated the script to factor this part indeed